virtcontainers/sandbox_test.go

```diff
@@ -1530,6 +1530,10 @@ func TestSandboxCreationFromConfigRollbackFromCreateSandbox(t *testing.T) {
 		Volumes:          nil,
 		Containers:       nil,
 	}
+
+	// Ensure hypervisor doesn't exist
+	assert.NoError(os.Remove(hConf.HypervisorPath))
+
 	_, err := createSandboxFromConfig(ctx, sConf, nil)
```

Comment:
Given we previously failed(correctly), but with the wrong error - should we be checking for the actual error we fail with as well iyswim - to avoid such 'failed, but for the wrong reason' problems recurring?

Reply:
Yeah, I do swym, but that would require string matching on an error string which is going to be fragile. The only place we do this afaik is in the mock package to assert we got the correct "type" of error:

- https://github.com/kata-containers/runtime/blob/master/virtcontainers/pkg/vcmock/utils.go#L26

For example,

- https://github.com/kata-containers/runtime/blob/master/virtcontainers/pkg/vcmock/mock_test.go#L153..L155

But you are right, we should consider in future updating the entire codebase to return a typed error, not just a simple string error as we can check for the specific error types in the tests. That would be a huge job though ;)

Alternatively, we could do something crazy like my "structured errors" idea: https://github.com/kata-containers/kata-containers/issues/16 ;)